src/fortran/rangen.fpp

```diff
@@ -1,380 +1,164 @@
-              IJ97     = IJ97 - 1
-              IF ( IJ97 .EQ. 0 ) IJ97 = 97
-              CC       = CC - CD
-              IF ( CC .LT. 0.D0 ) CC  = CC + CM
-            ENDDO
-          ENDDO
-          I97(JSEQ) = II97
-          J97(JSEQ) = IJ97
-          C(JSEQ)   = CC
-          DO  JJ = 1, 97
-            U(JJ,JSEQ) = UU(JJ)
-          ENDDO
-        ENDIF
-      ELSEIF ( INDEX(CCHOPT,'R') .NE. 0 ) THEN
-        IF ( ISEQ .GT. 0 ) THEN
-          JSEQ = ISEQ
-        ELSE
-          ISEQ = JSEQ
-        ENDIF
-        IF ( INDEX(CCHOPT,'V') .NE. 0 ) THEN
-          WRITE(IORNDM,'(3Z8)') IJKL(JSEQ),NTOT(JSEQ),NTOT2(JSEQ)
-          WRITE(IORNDM,'(2Z8,Z16)') I97(JSEQ),J97(JSEQ),C(JSEQ)
-          WRITE(IORNDM,'(24(4Z16,/),Z16)') U
-        ELSE
-          ISEED(1) = IJKL(JSEQ)
-          ISEED(2) = NTOT(JSEQ)
-          ISEED(3) = NTOT2(JSEQ)
-        ENDIF
-      ENDIF
-
-      RETURN
       END
 
-*-- Author :    D. HECK IK FZK KARLSRUHE   17/03/2003
-C=======================================================================
+      DOUBLE PRECISION FUNCTION gasdev( dummy )
+      implicit none
 
-      SUBROUTINE RMMARD( RVEC,LENV,ISEQ )
+      external npygas
+      integer dummy
+      integer*8 bitgen
+      COMMON /npy/bitgen
 
-C-----------------------------------------------------------------------
-C  R(ANDO)M (NUMBER GENERATOR OF) MAR(SAGLIA TYPE) D(OUBLE PRECISION)
-C
-C  THESE ROUTINES (RMMARD,RMMAQD) ARE MODIFIED VERSIONS OF ROUTINES
-C  FROM THE CERN LIBRARIES. DESCRIPTION OF ALGORITHM SEE:
-C   http://wwwasdoc.web.cern.ch/wwwasdoc/cernlib.html (v113)
-C  IT HAS BEEN CHECKED THAT RESULTS ARE BIT-IDENTICAL WITH CERN
-C  DOUBLE PRECISION RANDOM NUMBER GENERATOR RMM48, DESCRIBED IN
-C   http://wwwasdoc.web.cern.ch/wwwasdoc/cernlib.html (v116)
-C  ARGUMENTS:
-C   RVEC   = DOUBLE PREC. VECTOR FIELD TO BE FILLED WITH RANDOM NUMBERS
-C   LENV   = LENGTH OF VECTOR (# OF RANDNUMBERS TO BE GENERATED)
-C   ISEQ   = # OF RANDOM SEQUENCE
-C
-C  VERSION OF D. HECK FOR DOUBLE PRECISION RANDOM NUMBERS.
-C-----------------------------------------------------------------------
+      call NPYGAS(gasdev, bitgen)
 
-      IMPLICIT NONE
+      end
 
-      INTEGER          KSEQ
-      PARAMETER        (KSEQ = 8)
-      COMMON /CRRANMA3/CD,CINT,CM,TWOM24,TWOM48,MODCNS
-      DOUBLE PRECISION CD,CINT,CM,TWOM24,TWOM48
-      INTEGER          MODCNS
+      REAL FUNCTION spgasdev( dummy )
+      implicit none
 
-      COMMON /CRRANMA4/C,U,IJKL,I97,J97,NTOT,NTOT2,JSEQ
-      DOUBLE PRECISION C(KSEQ),U(97,KSEQ),UNI
-      INTEGER          IJKL(KSEQ),I97(KSEQ),J97(KSEQ),
-     *                 NTOT(KSEQ),NTOT2(KSEQ),JSEQ
+      external npygas
+      integer dummy
+      integer*8 bitgen
+      COMMON /npy/bitgen
+      double precision rval
 
-      DOUBLE PRECISION RVEC(*)
-      INTEGER          ISEQ,IVEC,LENV
-      SAVE
-C-----------------------------------------------------------------------
+      call NPYGAS(rval, bitgen)
```

Comment:
I noticed that the code in the file is mixture of upper and lower cases. Probably, it is better to convert it to the upper case.

Reply:
Fortran ignores case, so let's leave it as it is. Or convert everything to lowercase.

Reply:
Btw, I actually tried to format a little bit so that variables are lowercase and functions / subroutines are upper case. To make the code more readable than having one default case.

Reply:
I also prefer lower case. It is just the most of the code is written in upper case.

Reply:
The F77 standard is upper case. F90 is flexible. In DPMJet I standardized the case using a formatter (PlusFORT) a long while ago. Here, I'd say it's not important.